File: Lab6BankNew/src/main/java/com/example/lab6banknew/CardAccount.java
package com.example.lab6banknew;

import javafx.beans.property.SimpleDoubleProperty;
import javafx.beans.property.SimpleStringProperty;

import java.io.Serializable;

public class CardAccount implements Serializable {
    private SimpleStringProperty cardNumber;
    private SimpleStringProperty customerName;
    private SimpleStringProperty customerAddress;
    private SimpleDoubleProperty accountBalance;
    private SimpleStringProperty accountType;
    private SimpleStringProperty issueDate;
    private SimpleStringProperty expiryDate;

    public CardAccount(String cardNumber, String customerName, String customerAddress,
                       double accountBalance, String accountType, String issueDate, String expiryDate) {
        this.cardNumber = new SimpleStringProperty(cardNumber);
        this.customerName = new SimpleStringProperty(customerName);
        this.customerAddress = new SimpleStringProperty(customerAddress);
        this.accountBalance = new SimpleDoubleProperty(accountBalance);
        this.accountType = new SimpleStringProperty(accountType);
        this.issueDate = new SimpleStringProperty(issueDate);
        this.expiryDate = new SimpleStringProperty(expiryDate);
    }

    public String getCardNumber() {
        return cardNumber.get();
    }

    public SimpleStringProperty cardNumberProperty() {
        return cardNumber;
    }

    public String getCustomerName() {
        return customerName.get();
    }

    public SimpleStringProperty customerNameProperty() {
        return customerName;
    }

    public String getCustomerAddress() {
        return customerAddress.get();
    }

    public SimpleStringProperty customerAddressProperty() {
        return customerAddress;
    }

    public double getAccountBalance() {
        return accountBalance.get();
    }

    public SimpleDoubleProperty accountBalanceProperty() {
        return accountBalance;
    }

    public String getAccountType() {
        return accountType.get();
    }

    public SimpleStringProperty accountTypeProperty() {
        return accountType;
    }

    public String getIssueDate() {
        return issueDate.get();
    }

    public SimpleStringProperty issueDateProperty() {
        return issueDate;
    }

    public String getExpiryDate() {
        return expiryDate.get();
    }

    public SimpleStringProperty expiryDateProperty() {
        return expiryDate;
    }
}

File: Lab6Kotlin/src/main/kotlin/com/example/lab6kotlin/CardAccount.kt
package com.example.lab6kotlin

import javafx.beans.property.SimpleDoubleProperty
import javafx.beans.property.SimpleStringProperty
import java.io.Serializable

class CardAccount(cardNumber: String?, customerName: String?, customerAddress: String?,
                  accountBalance: Double, accountType: String?, issueDate: String?, expiryDate: String?) : Serializable {
    private val cardNumber: SimpleStringProperty
    private val customerName: SimpleStringProperty
    private val customerAddress: SimpleStringProperty
    private val accountBalance: SimpleDoubleProperty
    val accountType: SimpleStringProperty
    private val issueDate: SimpleStringProperty
    private val expiryDate: SimpleStringProperty

    init {
        this.cardNumber = SimpleStringProperty(cardNumber)
        this.customerName = SimpleStringProperty(customerName)
        this.customerAddress = SimpleStringProperty(customerAddress)
        this.accountBalance = SimpleDoubleProperty(accountBalance)
        this.accountType = SimpleStringProperty(accountType)
        this.issueDate = SimpleStringProperty(issueDate)
        this.expiryDate = SimpleStringProperty(expiryDate)
    }

    fun getCardNumber(): String {
        return cardNumber.get()
    }

    fun cardNumberProperty(): SimpleStringProperty {
        return cardNumber
    }

    fun getCustomerName(): String {
        return customerName.get()
    }

    fun customerNameProperty(): SimpleStringProperty {
        return customerName
    }

    fun getCustomerAddress(): String {
        return customerAddress.get()
    }

    fun customerAddressProperty(): SimpleStringProperty {
        return customerAddress
    }

    fun getAccountBalance(): Double {
        return accountBalance.get()
    }

    fun accountBalanceProperty(): SimpleDoubleProperty {
        return accountBalance
    }

    fun getAccountType(): String {
        return accountType.get()
    }

    fun accountTypeProperty(): SimpleStringProperty {
        return accountType
    }

    fun getIssueDate(): String {
        return issueDate.get()
    }

    fun issueDateProperty(): SimpleStringProperty {
        return issueDate
    }

    fun getExpiryDate(): String {
        return expiryDate.get()
    }

    fun expiryDateProperty(): SimpleStringProperty {
        return expiryDate
    }
}